RecoTracker/FinalTrackSelectors/plugins/TrackLwtnnClassifier.cc

```diff
@@ -0,0 +1,84 @@
+#include "RecoTracker/FinalTrackSelectors/interface/TrackMVAClassifier.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "getBestVertex.h"
+
+//from lwtnn
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "lwtnn/LightweightNeuralNetwork.hh"
+
+namespace {
+  struct lwtnn {
+    lwtnn(const edm::ParameterSet& cfg):
+      lwtnnLabel_(cfg.getParameter<std::string>("lwtnnLabel"))
+    {}
+
+    static const char *name() { return "TrackLwtnnClassifier"; }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
+      desc.add<std::string>("lwtnnLabel", "trackSelectionLwtnn");
+    }
+
+    void beginStream() {}
+    void initEvent(const edm::EventSetup& es) {
+      edm::ESHandle<lwt::LightweightNeuralNetwork> lwtnnHandle;
+      es.get<CkfComponentsRecord>().get(lwtnnLabel_, lwtnnHandle);
+      neuralNetwork_ = lwtnnHandle.product();
+    }
+
+    float operator()(reco::Track const & trk,
+                     reco::BeamSpot const & beamSpot,
+                     reco::VertexCollection const & vertices) {
+
+      Point bestVertex = getBestVertex(trk,vertices);
+
+      inputs_["trk_pt"] = trk.pt();
+      inputs_["trk_eta"] = trk.eta();
+      inputs_["trk_lambda"] = trk.lambda();
+      inputs_["trk_dxy"] = trk.dxy(beamSpot.position()); // is the training with abs() or not?
+      inputs_["trk_dz"] = trk.dz(beamSpot.position()); // is the training with abs() or not?
+      inputs_["trk_dxyClosestPV"] = trk.dxy(bestVertex); // is the training with abs() or not?
+      inputs_["trk_dzClosestPV"] = trk.dz(bestVertex); // is the training with abs() or not?
```

Comment:
are these questions in the code still relevant?
It would be best to resolve and drop these or change them to statements.

Reply:
trk_dz better be NOT abs as it can correlate with eta to identify the region of the detector crossed.

Reply:
In this version the training is done without anything being abs. I'll change these to statements in the next commit.